openpype/hosts/maya/plugins/publish/collect_render.py

```diff
@@ -69,388 +68,248 @@ class CollectMayaRender(pyblish.api.ContextPlugin):
+            expected_files.append(
+                {
+                    product_name: layer_render_products.get_files(
+                        product)
+                })
+
+        has_cameras = any(product.camera for product in render_products)
+        assert has_cameras, "No render cameras found."
+
+        self.log.info("multipart: {}".format(
+            multipart))
+        assert expected_files, "no file names were generated, this is a bug"
+        self.log.info(
+            "expected files: {}".format(
+                json.dumps(expected_files, indent=4, sort_keys=True)
             )
+        )
 
-            instance = context.create_instance(expected_layer_name)
-            instance.data["label"] = label
-            instance.data["farm"] = True
-            instance.data.update(data)
-
-    def parse_options(self, render_globals):
-        """Get all overrides with a value, skip those without.
-
-        Here's the kicker. These globals override defaults in the submission
-        integrator, but an empty value means no overriding is made.
-        Otherwise, Frames would override the default frames set under globals.
-
-        Args:
-            render_globals (str): collection of render globals
-
-        Returns:
-            dict: only overrides with values
-
-        """
-        attributes = lib.read(render_globals)
-
-        options = {"renderGlobals": {}}
-        options["renderGlobals"]["Priority"] = attributes["priority"]
-
-        # Check for specific pools
-        pool_a, pool_b = self._discover_pools(attributes)
-        options["renderGlobals"].update({"Pool": pool_a})
-        if pool_b:
-            options["renderGlobals"].update({"SecondaryPool": pool_b})
+        # if we want to attach render to subset, check if we have AOV's
+        # in expectedFiles. If so, raise error as we cannot attach AOV
+        # (considered to be subset on its own) to another subset
+        if attach_to:
+            assert isinstance(expected_files, list), (
+                "attaching multiple AOVs or renderable cameras to "
+                "subset is not supported"
+            )
 
-        # Machine list
-        machine_list = attributes["machineList"]
-        if machine_list:
-            key = "Whitelist" if attributes["whitelist"] else "Blacklist"
-            options["renderGlobals"][key] = machine_list
+        # append full path
+        aov_dict = {}
+        default_render_folder = context.data.get("project_settings")\
+            .get("maya")\
+            .get("RenderSettings")\
+            .get("default_render_image_folder") or ""
+        # replace relative paths with absolute. Render products are
+        # returned as list of dictionaries.
+        publish_meta_path = None
+        for aov in expected_files:
+            full_paths = []
+            aov_first_key = list(aov.keys())[0]
+            for file in aov[aov_first_key]:
```

Comment:
```suggestion
            for file_ in aov[aov_first_key]:
```

Reply:
Any reason behind this?

Reply:
oh my bad, I assumed it was a built-in type because VSCode displays it as a different color than other variables...
![image](https://user-images.githubusercontent.com/4348536/230214686-a638a8f5-424f-4543-853b-2f7982578adc.png)

Reply:
Ah, it's because I think it is a builtin in Py2: https://stackoverflow.com/a/24942363
I'll ignore it for now. To the future! 🚀